### PR TITLE
Fix AV in unittest when setting pointer to nullptr, we should never promote the AuxPtrs so remove the dependency on recycler on that case

### PR DESCRIPTION
**Pull request by leirocks**
Fix AV in unittest
when setting pointer to nullptr, we should never promote the AuxPtrs so remove the dependency on recycler on that case

### CODE REVIEW

File: lib/Runtime/Base/AuxPtrs.h [hunk 1]
```diff
@@ -165,15 +165,16 @@ namespace Js
     }
 
     template<class T, typename FieldsEnum>
-    void AuxPtrs<T, FieldsEnum>::AllocAuxPtrFix(T* _this, uint8 size, Recycler* recycler)
+    void AuxPtrs<T, FieldsEnum>::AllocAuxPtrFix(T* pHost, uint8 size, Recycler* recycler)
```
Comment (MikeHolman) [inline lib/Runtime/Base/AuxPtrs.h]: nit: imo, should just be `host`... `pHost` is pretty non-standard notation for our codebase.

Comment (leirocks) [inline lib/Runtime/Base/AuxPtrs.h]: OK, thanks!

**Comment by leirocks** [comment]
btw, another fix might be just set auxPtrs to nullptr instead of setting the pointers to null one by one. need to investigate if there's any side effect though. Use current fix to unblock test first

**Comment by MikeHolman** [comment]
>another fix might be just set auxPtrs to nullptr instead of setting the pointers to null one by one

@leirocks Yes I was just thinking about this. Prior to your change, were we nulling out all the pointers? In either case, I think your current change is sufficient to unblock testing.

**Comment by leirocks** [comment]
@MikeHolman from FunctionBody::CleanUp(), it's not nulling out all. but I'm not sure if there's pointer need to be accessed after that or just because those pointers won't cause circular reference and didn't null out them before. so need investigation and testing. I'm merging this change for now. thanks!

**Pull request merged**